lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -291,6 +293,81 @@ def add_level_3_data(post, options)
         post[:additionalData].compact!
       end
 
+      def add_data_airline(post, options)
+        return unless options[:additional_data_airline]
+
+        mapper = {
+          'airline.agency_invoice_number': 'agency_invoice_number',
+          'airline.agency_plan_name': 'agency_plan_name',
+          'airline.airline_code': 'airline_code',
+          'airline.airline_designator_code': 'airline_designator_code',
+          'airline.boarding_fee': 'boarding_fee',
+          'airline.computerized_reservation_system': 'computerized_reservation_system',
+          'airline.customer_reference_number': 'customer_reference_number',
+          'airline.document_type': 'document_type',
+          'airline.flight_date': 'flight_date',
+          'airline.ticket_issue_address': 'ticket_issue_address',
+          'airline.ticket_number ': 'ticket_number',
+          'airline.travel_agency_code': 'travel_agency_code',
+          'airline.travel_agency_name': 'travel_agency_name',
+          'airline.passenger_name': 'passenger_name'
+        }
+
+        post[:additionalData].merge!(extract_and_transform(mapper, options[:additional_data_airline]))
+
+        if options[:additional_data_airline][:leg].present?
+          leg_data = {
+            'airline.leg.carrier_code': 'carrier_code',
+            'airline.leg.class_of_travel': 'class_of_travel',
+            'airline.leg.date_of_travel': 'date_of_travel',
+            'airline.leg.depart_airport': 'depart_airport',
+            'airline.leg.depart_tax': 'depart_tax',
+            'airline.leg.destination_code': 'destination_code',
+            'airline.leg.fare_base_code': 'fare_base_code',
+            'airline.leg.flight_number': 'flight_number',
+            'airline.leg.stop_over_code': 'stop_over_code'
+          }
+
+          post[:additionalData].merge!(extract_and_transform(leg_data, options[:additional_data_airline][:leg]))
+        end
+
+        if options[:additional_data_airline][:passenger].present?
+          passenger_data = {
+            'airline.passenger.date_of_birth': 'date_of_birth',
+            'airline.passenger.first_name': 'first_name',
+            'airline.passenger.last_name': 'last_name',
+            'airline.passenger.telephone_number': 'telephone_number',
+            'airline.passenger.traveller_type': 'traveller_type'
+          }
+          post[:additionalData].merge!(extract_and_transform(passenger_data, options[:additional_data_airline][:passenger]))
+        end
+        post[:additionalData].compact!
```

Comment:
⚠️ Warning:

`compact!` is a tricky method if all the keys on the has had values it returns nil:

```bash
irb(main):001:0> {uno: 1}.compact!
=> nil
irb(main):002:0> {uno: 1, dos: nil}.compact!
=> {:uno=>1}
```

Reply:
I think that could be ok since we don’t want to save nil values